app/controllers/dois_controller.rb

```diff
@@ -14,7 +14,7 @@ def create
     # I don't think we need to do anything special here, just skip over the call
     # to MintDoiAsync and redirect back where we came from?
   ensure
-    redirect_to resource_path(params[:resource_id])
+    redirect_back fallback_location: resource_path(params[:resource_id])
```

Comment:
When a DOI is minted, return to wherever HTTP_REFERRER header is set to. Barring that, just go back to the resource page